### PR TITLE
[networks] Add support for DNS with NAT

### PR DESCRIPTION
**Pull request by p-lambert**
### What does this PR do?

Extend support for DNS stats in the context of NAT.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Describe how to test your changes

Change covered by integration test.

### Checklist

- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] The `need-change/operator` and `need-change/helm` labels has been applied if applicable.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated if applicable.

Note: Adding GitHub labels is only possible for contributors with write access.

### CODE REVIEW

File: pkg/network/tracer/tracer.go [hunk 1]
```diff
@@ -437,8 +437,8 @@ func (t *Tracer) getConnections(buffer []network.ConnectionStats) ([]network.Con
 		return nil, 0, err
 	}
 
-	for _, conn := range active {
-		conn.IPTranslation = t.conntracker.GetTranslationForConn(conn)
+	for i, conn := range active {
+		active[i].IPTranslation = t.conntracker.GetTranslationForConn(conn)
```
Comment (p-lambert) [inline pkg/network/tracer/tracer.go:441]: This is fixing a bug recently introduced but not yet released 😬

File: pkg/network/nat.go [hunk 2]
```diff
@@ -1,12 +1,11 @@
-package nat
+package network
```
Comment (p-lambert) [inline pkg/network/nat.go:1]: This is to prevent an import loop

File: pkg/network/tracer/tracer.go (hunk 1 above)
Comment (hmahmood) [reply pkg/network/tracer/tracer.go:441]: Can we get rid of `conn` altogether?

**Review by hmahmood — APPROVED** [review]

File: pkg/network/tracer/tracer.go (hunk 1 above)
Comment (brycekahle) [reply pkg/network/tracer/tracer.go:441]: We should probably add a test around these connections get translations, if this slipped through.

File: pkg/network/tracer/tracer_test.go [hunk 3]
```diff
@@ -1058,6 +1059,15 @@ func TestDNSStatsForTimeout(t *testing.T) {
 	testDNSStats(t, "golang.org", 0, 0, 1, "1.2.3.4")
 }
 
+func TestDNSStatsWithNAT(t *testing.T) {
+	// Setup a NAT rule to translate 2.2.2.2 to 8.8.8.8 and issue a DNS request to 2.2.2.2
+	cmds := []string{"iptables -t nat -A OUTPUT -d 2.2.2.2 -j DNAT --to-destination 8.8.8.8"}
+	nettestutil.RunCommands(t, cmds, true)
+	testDNSStats(t, "golang.org", 1, 0, 0, "2.2.2.2")
+	cmds = []string{"iptables -t nat -D OUTPUT -d 2.2.2.2 -j DNAT --to-destination 8.8.8.8"}
+	nettestutil.RunCommands(t, cmds, true)
```
Comment (brycekahle) [inline pkg/network/tracer/tracer_test.go:1068]: You probably want this in a `defer`, so it gets torn down regardless of test result.

Comment (p-lambert) [reply pkg/network/tracer/tracer_test.go]: Done

File: pkg/network/tracer/tracer.go (hunk 1 above)
Comment (p-lambert) [reply pkg/network/tracer/tracer.go:441]: I added a regression test

**Review by brycekahle — APPROVED** [review]

**Pull request merged**